src/components/Directions/DirectionsAutocomplete.tsx

```diff
@@ -146,6 +159,46 @@ export const DirectionsAutocomplete = ({ label, value, setValue }: Props) => {
   const { userSettings } = useUserSettingsContext();
   const { isImperial } = userSettings;
 
+  const ALPHABETICAL_MARKER = useMemo(() => {
+    let svgElement;
+    if (typeof window !== 'undefined' && typeof document !== 'undefined') {
+      svgElement = document.createElement('div');
+      svgElement.innerHTML = ReactDOMServer.renderToStaticMarkup(
+        <AlphabeticalMarker index={pointIndex} hasShadow width={27} />,
+      );
+    } else svgElement = undefined;
+
+    return {
+      color: 'salmon',
+      draggable: true,
+      element: svgElement,
+      offset: [0, -10] as PointLike,
+    };
+  }, [pointIndex]);
+
+  const markerRef = useRef<maplibregl.Marker>();
+
+  useEffect(() => {
+    const map = getGlobalMap();
+    if (value?.type === 'coords') {
+      markerRef.current = new maplibregl.Marker(ALPHABETICAL_MARKER)
+        .setLngLat(value.coords.center as LngLatLike)
+        .addTo(map);
+    }
+    return () => {
+      markerRef.current?.remove();
+    };
+  }, [ALPHABETICAL_MARKER, value]);
+
+  const onDragEnd = () => {
+    const lngLat = markerRef.current?.getLngLat();
+    if (lngLat) {
+      setValue(getCoordsOption([lngLat.lng, lngLat.lat]));
+    }
+  };
+
+  markerRef.current?.on('dragend', onDragEnd);
```

Comment:
I think this should be inside the useEffect above. Currently this would add the listener on every component render, that can degrade the app dramatically.